Lesson16_Task01.py

```diff
@@ -0,0 +1,126 @@
+"""
+Home-task 13 Class writing Practice
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes, as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters
+# do not forget about block '''if __name__ == '__main__':'''
+"""
+
+import datetime
+
+
+class BankCard:
+    """
+    This class represents a bank card with various attributes and operations.
+    """
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date):
+        self.card_num = card_num
+        self.cvv = cvv
+        self.pin = pin
+        self.name = name
+        self.surname = surname
+        self.end_date = end_date
+        self.is_blocked = False
+        self.balance = 0.0
+
+    @classmethod
+    def generate_expiry_date(cls):
```

Comment:
classmethod is method that uses only class attributes. For example PIN_SIZE, BANK_ID ( something that common for all issued card). Exactly in this case you have staticmethod -> method that does NOT use self or cls